### ISSUE TITLE
Investigate a DOC quick search feature.

### ISSUE THREAD

**Issue by samccone** [marionettejs/marionettejs.com#53]
We have all of the page contents at compile time.. so i think it should be possible to create some kind of quick search in our doc pages.

**Comment by kkemple** [comment]
i built something like this for my blog using lunr.js and backbone :: wrote a post about it here :: http://kkemple.github.io/adding-site-search-with-lunr/

**Comment by kkemple** [comment]
@samccone essentially you would have a grunt task that builds the json to feed to the collection, then you just fetch it at runtime and populate the the lunr search index with the collection data, very simple

**Comment by paulfalgout** [comment]
This definitely sounds like the right solution

**Comment by kkemple** [comment]
There is also Swiftype :: https://swiftype.com/open-source :: that gives free premium service to open source projects. I believe CodeShip, Ember, and SendGrid also use them for search, worth looking in to.

**Comment by thejameskyle** [comment]
@kkemple Just asked https://twitter.com/stefanpenner/status/551486435581501441

**Comment by samccone** [comment]
that stinks, 

we can always ask https://qbox.io/ if they will sponsor us.. and we can just use ES.. which is the correct tech for this problem

**Comment by kkemple** [comment]
yeah elastic search is good, but maybe it wouldn't hurt to give lunr.js a go just to perf it, if we can manage by ( categories | tags | keywords ) etc. to keep the json size down, or only populate once the user clicks search, we could get a much faster search in place, since we have to compile the docs anyway it may not hurt to use something local.

Also it would be better for offline, we could store the search collections in localStorage and just sync so there would only be requests when absolutely needed, that is the one side-effect of anything over an api, makes offline annoying. Also if we were to implement our own solution we could later add metrics gathering and then you could add a ranking system for most relevent to the search etc.. might be a pure case of YAGNI at this point, but always good to be ready for change :wink:

**Comment by samccone** [comment]
yeah @kkemple not opposed to this :) just throwing out ideas

**Comment by kkemple** [comment]
yeah well my idea may be overkill, it would be weird to build so much infrastructure around the search when the site is pretty static, we could let the ES service handle the work allowing the docs to stay dumber, if we went to an SPA :: #66 :: then it might make more sense for something local... I too am just trying to work through all the scenarios :smiley_cat:

**Comment by alexbrasetvik** [comment]
Hi.

I'm representing [Found.no](https://www.found.no), and [Sam asked me to chime in here](https://twitter.com/samccone/status/552222696960843776). 

We provide hosted Elasticsearch as a service, and we are very happy Marionette users. We're happy to provide you with the necessary clusters.

If you sign up, we can take it from there. :)

**Comment by peterblazejewicz** [comment]
As you maybe noticed we have been quite heavily indexed. Can we use Google Search Engine (we can customize their L&F)?
sample, just type `view` or `whatever`:
https://www.google.com/cse/publicurl?cx=014386862376114157370:n4_zeoqmnds
